src/main/kotlin/no/nav/familie/ef/sak/vilkår/regler/vilkår/AleneomsorgRegel.kt

```diff
@@ -45,28 +57,110 @@ class AleneomsorgRegel : Vilkårsregel(
         }
     }
 
-    private fun opprettAutomatiskBeregnetNæreBoforholdDelvilkår() = Delvilkårsvurdering(
-        resultat = Vilkårsresultat.AUTOMATISK_OPPFYLT,
-        listOf(
-            Vurdering(
-                regelId = RegelId.NÆRE_BOFORHOLD,
-                svar = SvarId.NEI,
-                begrunnelse = "Automatisk vurdert (${LocalDate.now().norskFormat()}): Det er beregnet at annen forelder bor mer enn 1 km unna bruker.",
+    private fun erDigitalSøknadOgDonorbarnEllerTerminbarnOgDonorBarnOgHarSammeAdresse(
+        metadata: HovedregelMetadata,
+        barnId: UUID?,
+    ): Boolean {
+        val søknadsgrunnlagBarn =
+            metadata.vilkårgrunnlagDto.barnMedSamvær.find {
+                it.barnId == barnId
+            }?.søknadsgrunnlag
+
+        val registergrunnlagBarn =
+            metadata.vilkårgrunnlagDto.barnMedSamvær.find {
+                it.barnId == barnId
+            }?.registergrunnlag
+
+        if (søknadsgrunnlagBarn == null || registergrunnlagBarn == null) return false
+
+        return erDigitalSøknad(metadata) &&
+            (
+                erDonorbarnOgHarSammeAdresse(
+                    søknadsgrunnlagBarn,
+                    registergrunnlagBarn,
+                ) || erTerminbarnOgErDonorbarnOgHarSammeAdresse(søknadsgrunnlagBarn)
```

Comment:
`erDigitalSøknad && erDonorbarn && (harSammeRegisterAdresse(..) || erTerminBarnMedSammeAdresse(..) )`


`erDonorbarn` er felles for begge de to funksjonene

Reply:
Godt poeng - endrer!

Reply:
fixd [e29b83a](https://github.com/navikt/familie-ef-sak/pull/2558/commits/e29b83a3b3898db34672b571a2201238ec9e0338)